File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0;
        int l = 0;
        int r = height.size()-1;
        
        while(l <= r){
            int area = min(height[l], height[r])*(r-l);
            ans = max(area, ans);
            
            if(height[l] <= height[r]) l++;
            else r--;
        }
        
        
        return ans;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxA = 0
        l = 0
        r = len(height)-1
        
        while(l <= r):
            maxA = max(maxA, min(height[l], height[r])*(r-l))
            if(height[l] < height[r]):
                l += 1
            else:
                r -= 1
                
        return maxA